core/src/main/java/org/infinispan/manager/impl/UnwrappingEmbeddedCacheManager.java

```diff
@@ -0,0 +1,46 @@
+package org.infinispan.manager.impl;
+
+import org.infinispan.Cache;
+import org.infinispan.manager.EmbeddedCacheManager;
+
+/**
+ * EmbeddedCacheManager used for cluster executor invocation so that caches are not wrapped with security
+ * since invoking via ClusterExecutor already requires ADMIN privileges
```

Comment:
I'd rather require only `EXEC` for the cluster executor and pass the security principal to the remote nodes, assuming it is possible.

Reply:
The `DistributedExecutor` didn't currently support this, I think we need a separate JIRA to handle this though, as this will add a lot more changes to `ClusterExecutor` I wasn't planning for this PR.

Reply:
Ok, please create a separate JIRA.

Reply:
Created https://issues.jboss.org/browse/ISPN-9914